### DISCUSSION TITLE
Opencart 4 - Details, Suggestions and the Future

### DISCUSSION THREAD

**Discussion by felipoantonoff** [opencart/opencart#11419] [Ideas]
Hello, I opened an issue about relevant bugs and other simpler Opencart 4 bugs: https://github.com/opencart/opencart/issues/11418
I did some quick tests and would like to suggest some changes for the future of Opencar and also to hear the suggestion of other colleagues.

After testing, I consider that Opencart 4 is not production-ready at the moment, even though it has excellent performance and features.

**That's why I would recommend keeping the focus on the Opencart 3 version and I suggest some ideas:**
  * Remove Opencart 4 from Opencart.com Home from the downloads and demo part, due to its current bugs
  * Leave Opencart 3.0.3.8 as the main and most indicated on the site and in the demo
  * Already on Github it might be interesting to rename Opencart 4 to beta or maybe some other tag
  * Be focused on fixing all the most serious and relevant Opencart 4 bugs
  * I saw in the project https://github.com/opencart/opencart/projects/6 some future tasks such as new theme, review only after purchased and other tasks, I would recommend only relaunching Opencart 4 after all relevant tasks for it are finished, like the new theme, because it will change the structure a lot, so it's better to be released only after done this big change
  * Analyze the return of OCMOD or another tool to replace it, we are still without a similar development tool and more examples of use of Events for cases where it is interesting
  * Improve Checkout, perhaps the ideal would be to use Vue.js or use a good Checkout as a basis, if necessary I can open the code of our Checkout Lite made in Vue.js and within the standard of Opencart, ideal is usually the registration and then a One Page style Checkout or with horizontal steps

I feel like Opencart 4 needs a few more weeks or maybe months of development, maybe it would be good to relaunch it after adding the desired features, bug fixes, various tests and being prepared to move all kinds of modules from the Opencart 3 standard to 4 it, so I would recommend it be switched back to Opencart 3.0.3.8 and be calmly worked on Opencart 4 and its improvements for a future new release of it in a few weeks or maybe a few months if need be.

If any of the suggestions is not possible, I would recommend that at least a fix for the most relevant bugs that could affect an order and after-sales be released as soon as possible, as it greatly affects the use of Opencart by new customers and also its image.

**Suggestions for the future of Opencart and process improvement:**
  * Roadmap focusing on the short and long term
  * More standardized releases eg 2 years for a new major release and about 4-6 months for subversions with relevant features and immediate release for bug fixes and other small details
  * Improvement in the testing process, as it has some details of Opencart 4 and others that could be noticed with simple tests of registration, purchase, order editing and etc, so I would recommend a greater focus on testing and maybe even having a team member for testing or asking the community, opening a topic or something like that for developers to test too before a release with a lot of change
  * Creation of a system similar to OCMOD to replace it, we need resources to change platform core files, without directly changing the files for security and maintenance reasons
  * Standardize the quality of the modules on the Marketplace, I recommend a better curation and also copyright, as unfortunately there are many developers who improperly take the work of another colleague
  * Better curation in Opencart Partners, unfortunately there are some who want to create their own modules/products and even change their brand name to be accepted again in Opencart
  * Curation in the use of the Opencart brand, there are many companies that use the Opencart name and sometimes the client may think that it is an official Opencart company, when they only use part of the name
  * Better documentation of the platform in the technical/development and administrative part
  * Show more Opencart use cases
  * Improve the Marketplace system, there are still many simple bugs to solve, if you want I do a type of consulting
  * Redesign the look of the Opencart website and theme
  * Easy Opencart upgrade

I am available to carry out tests in Opencart and if you wish to help with any part of its code. I have more than 10 years of experience with it, mainly in the creation of modules and integrations.

**Comment by nexadomain** [comment]
Fully Agreed with you

**Reply by felipoantonoff** [reply]
Good to know, thanks for the support, let's see if other colleagues leave their opinions too.

**Comment by TheCartpenter** [comment]
By re-integrating OCMOD, the same issues as before will be shown especially on the OC Marketplace; bunch of line conflicts.

**Reply by felipoantonoff** [reply]
The OCMOD/vQmod is just a tool, it even caches changes made to files, so great performance too.
We can't blame a tool for average or bad developers or maybe they don't know how to use the tool in the best way.

The Marketplace will continue to have bad and conflicting modules with or without OCMOD/vQmod, until it has a better curation of products, developers and partners.

I also covered other subjects, OCMOD can be dealt with in this topic https://github.com/opencart/opencart/discussions/10883, which has the point of view of several developers.

**Comment by TheCartpenter** [comment]
The subject has been covered before.

**Reply by felipoantonoff** [reply]
Yes, if it is about OCMOD it was addressed, but there are also other subjects and even today 1 year later we do not have a natively substitute tool, the topic was more relevant in 2022 due to the launch of Opencart 4 without being ready for production.

I also summarized some suggestions and possible tips for the future of Opencart in order to exchange information with colleagues and make the Opencart project more future-proof and better for everyone.

**Comment by totalsport-tech** [comment]
https://totalsport.ua/futbol-c-31/futbolnaja-forma/
